lib/elixir/lib/calendar/iso.ex

```diff
@@ -1455,6 +1455,134 @@ defmodule Calendar.ISO do
     {days, {@parts_per_day - 1, @parts_per_day}}
   end
 
+  @doc """
+  Shifts date by Calendar.Duration units according to its calendar.
+
+  Available units are: `:year, :month, :week, :day, :hour, :minute, :second, :microsecond`.
+
+  When shifting by month:
+  - it will shift to the current day of a month
+  - when the current day does not exist in a month, it will shift to the last day of a month
+
+  ## Examples
+
+      iex> Calendar.ISO.shift_date(2016, 1, 3, month: 2)
+      {2016, 3, 3}
+      iex> Calendar.ISO.shift_date(2016, 2, 29, month: 1)
+      {2016, 3, 29}
+      iex> Calendar.ISO.shift_date(2016, 1, 31, month: 1)
+      {2016, 2, 29}
+      iex> Calendar.ISO.shift_date(2016, 1, 31, year: 4, day: 1)
```

Comment:
Nitpick. Let's pass the whole `Duration` struct here. We don't need to convert it to a keyword list before.